public/app/plugins/panel/graph/graph.ts

```diff
@@ -384,6 +386,57 @@ function graphDirective(timeSrv, popoverSrv, contextSrv) {
         return series;
       }
 
+      function highlightSeries(options) {
+        if (ctrl.highlighting === undefined) {
+          return;
+        }
+
+        for (let i = 0; i < sortedSeries.length; i++) {
+          if (ctrl.highlighting !== sortedSeries[i].alias) {
+            continue;
+          }
+
+          let highlightWidth = 0;
+          if (options.series.lines.show || sortedSeries[i].lines.show) {
+            highlightWidth =
+              ('lineWidth' in sortedSeries[i].lines
+                ? sortedSeries[i].lines.lineWidth
+                : options.series.lines.lineWidth) + 1;
+            if (highlightWidth > 1) {
+              sortedSeries[i].lines.lineWidth = highlightWidth;
+            }
+          }
+
+          if (options.series.lines.show || sortedSeries[i].dashes.show) {
+            highlightWidth =
+              ('lineWidth' in sortedSeries[i].dashes
+                ? sortedSeries[i].dashes.lineWidth
+                : options.series.lines.lineWidth) + 1;
+            if (highlightWidth > 1) {
+              sortedSeries[i].dashes.lineWidth = highlightWidth;
+            }
+          }
+
+          if (options.series.bars.show || sortedSeries[i].bars.show) {
```

Comment:
This is a nice feature. Great work.

But I don't think this effect on the bars is that useful. For me when I hover the legend the change to the bars are so small I can't se them. So for now I think we can remove highlight from the bars.

Reply:
It is easier to pull down than to build.